File: linearSearch.java
public class linearSearch {
    public static int LinearSearch(int arr[],int key,int i){
	if(i==arr.length){
		return -1;
	}
	if(arr[i]==key){
		return i;
	}
	return LinearSearch(arr,key,i+1);
}

 public static void main(String[] args) {
int arr[] = {10,45,23,64,12};
int key = 64;
int i = LinearSearch(arr,key,0);
if(i==-1){
	System.out.println("Element not found");
}else{
	System.out.println("Element found at index:"+i);
}
}
}

File: linearsearch.py
def linear_search(arr, target, index):
    if index == len(arr):
        return -1
    
    if arr[index] == target:
        return index
    
    return linear_search(arr, target, index + 1)


arr = [10, 20, 30, 40, 50]

target = int(input("Enter element to search: "))

result = linear_search(arr, target, 0)

if result == -1:
    print("Element not found")
else:
    print("Element found at index", result)